crypto/hpke/hpke.c

```diff
@@ -0,0 +1,2579 @@
+                     unsigned int mode, OSSL_HPKE_SUITE suite,
+                     const unsigned char *ikm, size_t ikmlen,
+                     unsigned char *pub, size_t *publen,
+                     EVP_PKEY **priv)
+{
+    return hpke_kg_evp(libctx, propq, mode, suite,
+                       ikmlen, ikm, publen, pub, priv);
+}
+
+/**
+ * @brief check if a suite is supported locally
+ *
+ * @param suite is the suite to check
+ * @return 1 for good/supported, not-1 otherwise
+ */
+int OSSL_HPKE_suite_check(OSSL_HPKE_SUITE suite)
+{
+    return hpke_suite_check(suite);
+}
+
+/*
+ * @brief get a (possibly) random suite, public key and ciphertext for GREASErs
+ *
+ * As usual buffers are caller allocated and lengths on input are buffer size.
+ *
+ * @param libctx is the context to use
+ * @param propq is a properties string
+ * @param suite_in specifies the preferred suite or NULL for a random choice
+ * @param suite is the chosen or random suite
+ * @param pub a random value of the appropriate length for a sender public value
+ * @param pub_len is the length of pub (buffer size on input)
+ * @param cipher is a random value of the appropriate length for a ciphertext
+ * @param cipher_len is the length of cipher
+ * @return 1 for success, otherwise failure
+ */
+int OSSL_HPKE_good4grease(OSSL_LIB_CTX *libctx, const char *propq,
+                          OSSL_HPKE_SUITE *suite_in,
+                          OSSL_HPKE_SUITE *suite,
+                          unsigned char *pub,
+                          size_t *pub_len,
+                          unsigned char *cipher,
+                          size_t cipher_len)
+{
+    return hpke_good4grease(libctx, propq, suite_in, suite,
+                            pub, pub_len, cipher, cipher_len);
+}
+
+/*
+ * @brief map a string to a HPKE suite
+ * @param str is the string value
+ * @param suite is the resulting suite
+ * @return 1 for success, otherwise failure
+ */
+int OSSL_HPKE_str2suite(const char *str, OSSL_HPKE_SUITE *suite)
+{
+    return hpke_str2suite(str, suite);
+}
+
+/*
+ * @brief tell the caller how big the cipertext will be
+ * @param suite is the suite to be used
+ * @param enclen points to what'll be enc length
+ * @param clearlen is the length of plaintext
+ * @param cipherlen points to what'll be ciphertext length
+ * @return 1 for success, otherwise failure
+ *
+ * AEAD algorithms add a tag for data authentication.
+ * Those are almost always, but not always, 16 octets
+ * long, and who know what'll be true in the future.
+ * So this function allows a caller to find out how
+ * much data expansion they'll see with a given
+ * suite.
+ */
+int OSSL_HPKE_expansion(OSSL_HPKE_SUITE suite,
+                        size_t *enclen,
+                        size_t clearlen,
+                        size_t *cipherlen)
+{
+    return hpke_expansion(suite, enclen, clearlen, cipherlen);
+}
+
+/* the "legacy" enc/dec API functions below here. will likely disappear */
```

Comment:
not sure TBH - I'd argue to keep 'em but not that strongly (unless I turn up something hard when ripping out that code from my ECH branch, which I don't expect) - will get back when that's attempted

Reply:
I asked the one other person I know who has played with those "legacy" APIs a few weeks ago if they still needed 'em and haven't heard back, so I guess it's ok to let 'em head off to wherever unused APIs go to retire:-) Will leave this open until I have the ECH build working properly though in case anything turns up.